modules/querier/querier_query_range.go

```diff
@@ -32,13 +32,16 @@ func (q *Querier) queryRangeRecent(ctx context.Context, req *tempopb.QueryRangeR
 		return nil, fmt.Errorf("error finding generators in Querier.queryRangeRecent: %w", err)
 	}
 
-	c, err := traceql.QueryRangeCombinerFor(req, traceql.AggregateModeSum)
+	c, err := traceql.QueryRangeCombinerFor(req, traceql.AggregateModeSum, q.cfg.Metrics.MaxResponseSeries)
 	if err != nil {
 		return nil, err
 	}
 
 	mtx := sync.Mutex{} // combiner doesn't lock, so take lock before calling Combine to make is safe
 	forEach := func(ctx context.Context, client tempopb.MetricsGeneratorClient) error {
+		if c.MaxSeriesReached() {
```

Comment:
should we not enforce this in the generators to prevent `{} | rate() by (span:id)` or whatever from overwhelming them?

have you tested such a query on this branch?

Reply:
also, don't we need similar code on the backend path? we also need to be thoughtful about the situation where the output series are different than the intermediate series.

for instance a `quantile_over_time()` calculation will pass up intermediate histograms that are then turned into quantiles in the frontend. so the queriers/generators may actually be handling more series than the output result.  i'm wondering if we want a 2 tier limit where the queriers/generators that are doing the intermediate work have a higher limit than the frontend? can you push this discussion internally?

Reply:
With the new approach, we are setting the same limit at the query-frontend level all the way down to the metrics evaluator level - this means at the default limit of 1000 - each intermediate step - each block would have 1000 series limit, each metrics-generator would have 1000 series limit before the responses converge at the query-frontend